lib/simulator-management.js

```diff
@@ -19,6 +19,12 @@ async function getExistingSim (opts) {
       return await getSimulator(device.udid);
     }
   }
+
+  for (let device of _.values(devices)) {
```

Comment:
This is good. Perhaps a log line, saying that the exact device was not found.

Alternatively, the loop could be performed a single time
```js
async function getExistingSim (opts) {
  const devices = await getDevices(opts.platformVersion);
  const appiumTestDeviceName = `appiumTest-${opts.deviceName}`;
  let appiumTestDevice;
  for (const device of _.values(devices)) {
    if (device.name === opts.deviceName) {
      return await getSimulator(device.udid);
    }
    if (device.name === appiumTestDeviceName) {
      appiumTestDevice = device;
    }
  }

  if (appiumTestDevice) {
    log.warn(`Unable to find device '${opts.deviceName}'. Found '${appiumTestDevice.name}' (udid: '${appiumTestDevice.udid}') instead`);
    return await getSimulator(appiumTestDevice.udid);
  }

  return null;
}
```

Reply:
👍 your code looks better. (I also came up with how to reduce loop flow.)